charts/capsule/Chart.yaml

```diff
@@ -21,8 +21,15 @@ sources:
 
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
-version: 0.1.9
+version: 0.2.0
 
 # This is the version number of the application being deployed.
 # This version number should be incremented each time you make changes to the application.
 appVersion: 0.1.1
+
+# Capsule Proxy Depndency
+dependencies:
```

Comment:
Yes, when the chart is built there will be a `Chart.lock` and the tar of the dependency in the `charts` folder (within the chart archive of capsule. I explicit do not push these thing (also include it in the `.gitignore`).  The lock looks like this:

```
dependencies:
- name: capsule-proxy
  repository: https://clastix.github.io/charts
  version: 0.2.4
digest: sha256:036f1908a3991a64c44eefe333357de55f4f7328dcc34edf61c4e7773bf3b76e
generated: "2022-06-16T20:51:19.608902+02:00"
```
I will add the Chart lock since it adds the feature of verifying the digest in the lock, but the charts should be pushed and only included in the chart archive.

Reply:
I recall we had a discussion with @bsctl in putting the Proxy as a dependency of Capsule, specifically in #231.

Honestly, I'm not sure about putting this as a dependency in the Helm installation, since there's not a hard requirement on the functionalities of Capsule without the Proxy, and not vice versa.

I see there's a condition to enable it.
https://github.com/clastix/capsule/blob/af32270120bd24f5a5c1e5b439fcf92eb6bacbc5/charts/capsule/Chart.yaml#L35

Let's try to summon the maintainers on this topic.

Reply:
Sure, My intention to make users more aware of the proxy project, because when we started using capsule, it took some time for me to understand, that there's a dedicated project for the proxy. Also I think the documentation does not give that hint: https://capsule.clastix.io/docs/general/proxy

Reply:
I see the point @oliverbaehler, and I think we need to improve the guidance towards Capsule Proxy, as it provides important features.
Anyway, I'd avoid to put it as a dependency simply because is not required. E.g. simpler scenario where straight API server is enough would be forced to have Capsule Proxy in place. I see instead the reverse order to be required: Capsule Proxy depends on Capsule.

BTW, I don't have a strong opinion. For me, it would be applicable @prometherion if we would like to want to (re-)define Capsule as a single and complete package to manage multi-tenancy soft-way. And this package would consist of Capsule+Capsule Proxy.

Reply:
In any case @oliverbaehler @prometherion this dependency is optional (see `dependencies.condition`), so IMO for me is perfect.

Reply:
Yeah I mean it doesn't change much just gives you the option to, by default it's disabled. So all the same.

Reply:
@oliverbaehler thanks for putting such effort on improving Capsule. I would avoid to create a dependency even if disabled by default. We’re planning Capsule EE as integrated stack of all the components. For the OSS version, I prefer to have independent modules. For sure we can improve the documentation to make users aware of the Capsule Proxy.

Reply:
if that's how u want it..

Reply:
In this case i will agree with @bsctl. In is better to inform users about the existence of capsule-proxy and additional features that they will get by installing it than adding it as a dependency in capsule helm chart, because capsule can work without it